proc-macros/src/rpc_macro.rs

```diff
@@ -48,12 +48,13 @@ pub struct RpcMethod {
 	pub returns: Option<syn::Type>,
 	pub signature: syn::TraitItemMethod,
 	pub aliases: Vec<String>,
+	pub with_context: bool,
 }
 
 impl RpcMethod {
 	pub fn from_item(attr: Attribute, mut method: syn::TraitItemMethod) -> syn::Result<Self> {
-		let [aliases, blocking, name, param_kind] =
-			AttributeMeta::parse(attr)?.retain(["aliases", "blocking", "name", "param_kind"])?;
+		let [aliases, blocking, name, param_kind, with_context] =
+			AttributeMeta::parse(attr)?.retain(["aliases", "blocking", "name", "param_kind", "with_context"])?;
```

Comment:
so, `with_context` shouldn't be needed if we decide to expose connection_id  globally...